### PR TITLE
Add unit tests for SubquerySegmentBinder

### LINKED ISSUE #28507: Add more unit tests for SubquerySegmentBinder
Hi community,

This issue is for https://github.com/apache/shardingsphere/discussions/28487.

## Aim
Add more unit tests for SubquerySegmentBinder to test its public functions to improve unit test coverage.

Unit test coverage can be viewed at: https://app.codecov.io/gh/apache/shardingsphere

## Basic Qualifications
- Java
- Maven
- Junit

### PR DESCRIPTION
**Pull request by HAibiiin**
Fixes #28507.

Changes proposed in this pull request:
  -
Add unit tests for SubquerySegmentBinder.java
---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [x] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        SubquerySegment subquerySegment = new SubquerySegment(39, 103, mySQLSelectStatement, "order_id = (SELECT order_id FROM t_order WHERE status = 'SUBMIT')");
+        SQLStatementBinderContext sqlStatementBinderContext =
+                new SQLStatementBinderContext(createMetaData(), DefaultDatabase.LOGIC_NAME, TypedSPILoader.getService(DatabaseType.class, "FIXTURE"), Collections.emptySet());
+        ColumnSegment boundedNameColumn = new ColumnSegment(7, 13, new IdentifierValue("user_id"));
+        boundedNameColumn.setColumnBoundedInfo(new ColumnSegmentBoundedInfo(new IdentifierValue(DefaultDatabase.LOGIC_NAME), new IdentifierValue(DefaultDatabase.LOGIC_NAME),
+                new IdentifierValue("t_order_item"), new IdentifierValue("user_id")));
+        sqlStatementBinderContext.getExternalTableBinderContexts().put("t_order_item", new SimpleTableSegmentBinderContext(Collections.singleton(new ColumnProjectionSegment(boundedNameColumn))));
+        Map<String, TableSegmentBinderContext> outerTableBinderContexts = new LinkedHashMap<>();
+        SubquerySegment actual = SubquerySegmentBinder.bind(subquerySegment, sqlStatementBinderContext, outerTableBinderContexts);
+
```
Comment (strongduanmu) [inline infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java:85]: Please remove useless blank line.

File: infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+
+public class SubquerySegmentBinderTest {
+    
+    /**
```
Comment (strongduanmu) [inline infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java:63]: Please remove javadoc in unit test.

File: infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+
+public class SubquerySegmentBinderTest {
```
Comment (strongduanmu) [inline infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java:61]: Please remove public modifier.

File: infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    
+    /**
+     * Test common scenarios.
+     */
+    @Test
+    public void assertBind() {
```
Comment (strongduanmu) [inline infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java:67]: Please remove public modifier.

File: infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        assertThat(((ColumnProjectionSegment) column).getColumn().getIdentifier().getValue(), is("order_id"));
+        assertNotNull(((ColumnProjectionSegment) column).getColumn().getColumnBoundedInfo());
+        assertThat(((ColumnProjectionSegment) column).getColumn().getColumnBoundedInfo().getOriginalColumn().getValue(), is("order_id"));
+        assertThat(((ColumnProjectionSegment) column).getColumn().getColumnBoundedInfo().getOriginalTable().getValue(), is("t_order"));
+        assertThat(((ColumnProjectionSegment) column).getColumn().getColumnBoundedInfo().getOriginalSchema().getValue(), is(DefaultDatabase.LOGIC_NAME));
+        assertThat(((ColumnProjectionSegment) column).getColumn().getColumnBoundedInfo().getOriginalDatabase().getValue(), is(DefaultDatabase.LOGIC_NAME));
+    }
+    
+    /**
+     * Test the 'With' clause scenario.
```
Comment (strongduanmu) [inline infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java:114]: Please remove javadoc in unit test.

File: infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+     */
+    @Test
+    public void assertBindUseWithClause() {
```
Comment (strongduanmu) [inline infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java:117]: Please remove public modifier.

File: infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+        WithSegment withSegment = new WithSegment(0, 74, Collections.singleton(commonTableExpressionSegment));
+        OracleSelectStatement oracleSelectStatement = new OracleSelectStatement();
+        oracleSelectStatement.setWithSegment(withSegment);
+        oracleSelectStatement.setProjections(new ProjectionsSegment(0, 0));
+        SubquerySegment subquerySegment = new SubquerySegment(0, 74, oracleSelectStatement, "WITH submit_order AS (SELECT order_id FROM t_order WHERE status = 'SUBMIT')");
+        SQLStatementBinderContext sqlStatementBinderContext =
+                new SQLStatementBinderContext(createMetaData(), DefaultDatabase.LOGIC_NAME, TypedSPILoader.getService(DatabaseType.class, "FIXTURE"), Collections.emptySet());
+        Map<String, TableSegmentBinderContext> outerTableBinderContexts = new LinkedHashMap<>();
+        SubquerySegment actual = SubquerySegmentBinder.bind(subquerySegment, sqlStatementBinderContext, outerTableBinderContexts);
+
```
Comment (strongduanmu) [inline infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java:137]: Please remove useless blank line.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java (hunk 1 above)
Comment (HAibiiin) [reply infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java]: Have been modified.

File: infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java (hunk 2 above)
Comment (HAibiiin) [reply infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java]: Have been modified.

File: infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java (hunk 3 above)
Comment (HAibiiin) [reply infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java]: Have been modified.

File: infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java (hunk 4 above)
Comment (HAibiiin) [reply infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java]: Have been modified.

File: infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java (hunk 5 above)
Comment (HAibiiin) [reply infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java]: Have been modified.

File: infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java (hunk 6 above)
Comment (HAibiiin) [reply infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java]: Have been modified.

File: infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java (hunk 7 above)
Comment (HAibiiin) [reply infra/binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/expression/impl/SubquerySegmentBinderTest.java]: Have been modified.

**Review by strongduanmu — APPROVED** [review]
Thank you for your contribution, merged.

**Pull request merged**